EEPROM class is throwing a warning: converting to non-pointer type 'nvs_handle {aka unsigned int}' from NULL

### Hardware:
Board:							ESP32 Dev Module
Core Installation version:			1.0.3-rc2
IDE name:						Arduino IDE
Flash Frequency:					40Mhz
PSRAM enabled:                                    no
Upload Speed:					115200
Computer OS:                                        Windows 10

### Description:
The constructor for the EEPROM class is throwing a warning when compiling:

```
C:\Users\User\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.3-rc2\libraries\EEPROM\src\EEPROM.cpp: In constructor 'EEPROMClass::EEPROMClass()':

C:\Users\User\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.3-rc2\libraries\EEPROM\src\EEPROM.cpp:37:25: warning: converting to non-pointer type 'nvs_handle {aka unsigned int}' from NULL [-Wconversion-null]

   , _user_defined_size(0)

                         ^

C:\Users\User\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.3-rc2\libraries\EEPROM\src\EEPROM.cpp: In constructor 'EEPROMClass::EEPROMClass(uint32_t)':

C:\Users\User\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.3-rc2\libraries\EEPROM\src\EEPROM.cpp:48:25: warning: converting to non-pointer type 'nvs_handle {aka unsigned int}' from NULL [-Wconversion-null]

   , _user_defined_size(0)

                         ^

C:\Users\User\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.3-rc2\libraries\EEPROM\src\EEPROM.cpp: In constructor 'EEPROMClass::EEPROMClass(const char*, uint32_t)':

C:\Users\User\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.3-rc2\libraries\EEPROM\src\EEPROM.cpp:58:41: warning: converting to non-pointer type 'nvs_handle {aka unsigned int}' from NULL [-Wconversion-null]

   , _user_defined_size(user_defined_size)
```

--- a/libraries/EEPROM/src/EEPROM.cpp
+++ b/libraries/EEPROM/src/EEPROM.cpp
@@ -29,7 +29,7 @@
 #include <esp_log.h>
 
 EEPROMClass::EEPROMClass(void)
-  : _handle(NULL)
+  : _handle(0)
   , _data(0)
   , _size(0)
   , _dirty(false)
@@ -40,7 +40,7 @@ EEPROMClass::EEPROMClass(void)
 
 EEPROMClass::EEPROMClass(uint32_t sector)
 // Only for compatiility, no sectors in nvs!
-  : _handle(NULL)
+  : _handle(0)
   , _data(0)
   , _size(0)
   , _dirty(false)
@@ -50,7 +50,7 @@ EEPROMClass::EEPROMClass(uint32_t sector)
 }
 
 EEPROMClass::EEPROMClass(const char* name, uint32_t user_defined_size)
-  : _handle(NULL)
+  : _handle(0)
   , _data(0)
   , _size(0)
   , _dirty(false)
@@ -60,7 +60,7 @@ EEPROMClass::EEPROMClass(const char* name, uint32_t user_defined_size)
 }
 
 EEPROMClass::~EEPROMClass() {
-  // end();
+  end();
 }
 
 bool EEPROMClass::begin(size_t size) {
@@ -152,6 +152,9 @@ void EEPROMClass::end() {
   }
   _data = 0;
   _size = 0;
+
+  nvs_close(_handle);
+  _handle = 0;
 }
 
 uint8_t EEPROMClass::read(int address) {

